backends/redis.go

```diff
@@ -1,109 +1,114 @@
+	return "Key not found"
+}
 
-	maxActive := 500
-	if config.MaxActive > 0 {
-		maxActive = config.MaxActive
-	}
+func (r *RedisBackend) newRedisClusterPool() redis.UniversalClient {
+	redisLogger.Info("Creating new Redis connection pool")
 
-	if config.EnableCluster {
-		redisLogger.Info("--> Using clustered mode")
-	}
+	timeout := 5 * time.Second
 
-	thisPoolConf := rediscluster.PoolConfig{
-		MaxIdle:       maxIdle,
-		MaxActive:     maxActive,
-		IdleTimeout:   240 * time.Second,
-		Database:      config.Database,
-		Password:      config.Password,
-		IsCluster:     config.EnableCluster,
-		UseTLS:        config.UseSSL,
-		TLSSkipVerify: config.SSLInsecureSkipVerify,
+	if r.config.Timeout > 0 {
+		timeout = time.Duration(r.config.Timeout) * time.Second
 	}
 
-	seed_redii := []map[string]string{}
+	var tlsConfig *tls.Config
+	if r.config.UseSSL {
+		tlsConfig = &tls.Config{
+			InsecureSkipVerify: r.config.SSLInsecureSkipVerify,
+		}
+	}
 
-	if len(config.Hosts) > 0 {
-		for h, p := range config.Hosts {
-			seed_redii = append(seed_redii, map[string]string{h: p})
+	var address []string
+	if len(r.config.Hosts) > 0 {
+		for h, p := range r.config.Hosts {
+			addr := h + ":" + p
+			address = append(address, addr)
 		}
 	} else {
-		redisLogger.Fatal("No Redis hosts set!")
+		addr := r.config.Host + ":" + strconv.Itoa(r.config.Port)
+		address = append(address, addr)
 	}
 
-	thisInstance := rediscluster.NewRedisCluster(seed_redii, thisPoolConf, false)
+	if !r.config.EnableCluster {
+		address = address[:1]
+	}
 
-	redisClusterSingleton = &thisInstance
+	var client redis.UniversalClient
+	opts := &RedisOpts{
+		MasterName:   r.config.MasterName,
+		Addrs:        address,
+		DB:           r.config.Database,
+		Password:     r.config.Password,
+		IdleTimeout:  240 * time.Second,
+		ReadTimeout:  timeout,
+		WriteTimeout: timeout,
+		DialTimeout:  timeout,
+		TLSConfig:    tlsConfig,
+	}
 
-	return &thisInstance
-}
+	if opts.MasterName != "" {
+		redisLogger.Info("Creating sentinel-backed fail-over client")
+		client = redis.NewFailoverClient(opts.failover())
+	} else if r.config.EnableCluster {
+		redisLogger.Info("Creating cluster client")
+		client = redis.NewClusterClient(opts.cluster())
+	} else {
+		redisLogger.Info("Creating single-node client")
+		client = redis.NewClient(opts.simple())
+	}
 
-func (r *RedisBackend) fixKey(keyName string) string {
-	return r.KeyPrefix + keyName
+	return client
 }
 
-func (r *RedisBackend) connect() {
-	if r.db == nil {
-		redisLogger.Debug("Connecting to redis")
-		r.db = newRedisClusterPool(false, r.config)
-	}
+func (r *RedisBackend) Connect() bool {
+	r.dbMu.Lock()
+	defer r.dbMu.Unlock()
 
-	redisLogger.Debug("Storage Engine already initialised...")
-	redisLogger.Debug("Redis handles: ", len(r.db.Handles))
+	r.db = r.newRedisClusterPool()
+	return true
+}
 
-	// Reset it just in case
-	r.db = redisClusterSingleton
+func (r *RedisBackend) cleanKey(keyName string) string {
```

Comment:
cleanKey function was added but is not used anywhere in the file

Reply:
done!